projects/npm-tools/packages/npm-scripts/src/sass/build.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const globby = require('globby');
+const r2 = require('liferay-r2');
+const path = require('path');
+const sass = require('sass');
+
+const log = require('../utils/log');
+
+const SASS_EXCLUDE = [
+	'**/_*.scss',
+	'**/build/**',
+	'**/classes/**',
+
+	// Copy from CSSBuilderArgs.java
+
+	'**/_diffs/**',
+	'**/.sass-cache*/**',
+	'**/.sass_cache_*/**',
+	'**/_sass_cache_*/**',
+	'**/_styled/**',
+	'**/_unstyled/**',
+	'**/css/aui/**',
+	'**/css/clay/**',
+	'**/tmp/**',
+];
+
+const getClayPaths = () => {
+	try {
+		return require('@clayui/css').includePaths;
+	} catch (e) {
+		return [];
+	}
+};
```

Comment:
This is a little weird. Might be some other ways to do this, but this was my first thought.

Reply:
What we have right now is that we allow passing `includePaths` to sass. So, we'd want https://github.com/liferay/liferay-frontend-projects/pull/245/files#diff-f3abf7ec1699e822b3182c10c3315fa3bd6516406a6c299b82c5b27f6d99155fR63 to be a configurable param.

That way, we could rely on whatever `@clayui/css` version is in `liferay-portal` at the time and not in what `npm-tools` is bringing over.

Reply:
> That way, we could rely on whatever @clayui/css version is in liferay-portal at the time and not in what npm-tools is bringing over.

@wincent,I didn't want to specify clay as a dependency in this package since we would have to release with every new clay version. By having this check, we protect from test failures without having to have clay as a dependency. We really wouldn't expect this to ever fail in the context of liferay-portal(only place this is used), but by doing this we don't require clay as a dependency in this package.

Not sure if this is the best way to go about it, but it made sense to me at the time.

Reply:
Having it as a `devDependency` like you do in this PR seems ok, especially if you make the version range pretty loose (makes it easier to update with `yarn upgrade-interactive` — and if all you need is to have it there for the purposes of running the tests, it wouldn't even need to be kept up-to-date as far as I can tell); as a dev dep, the actual version used here won't influence DXP at all.

The other option if it's just for testing is to create a scratch directory under `os.tmpdir()` and install Clay there from the network during the test run. Would be slower if you did it in a `beforeEach()`, but probably not in a `beforeAll()` (because CI is either installing it once as part of initial `yarn` set-up, or once as part of test run).

Reply:
I'm actually just going to remove the clay stuff and instead test against scss using `imports` and making my own file to include variables. 

I'm removing clay because its not always included when portal is building scss files. Any module that wants to use clay-css, references it directly via

```
buildCSS {
	imports = [
		new File(npmInstall.nodeModulesDir, "@clayui/css/src/scss")
	]
}
```